File: Sorting/75_sort_colours.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int count[] = {0, 0, 0};

        for(int num: nums) {
            count[num] += 1;
        }

        int left = 0;
        for(int i = 0; i <= 2; i += 1) {
            for(int j = 0; j < count[i]; j += 1) {
                nums[left++] = i;
            }
        }
    }
};

File: Sorting/75_sort_colours.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        hashmap = {}

        for num in nums:
            hashmap[num] = hashmap.get(num, 0) + 1

        index = 0
        for i in range(3):
            for j in range(hashmap.get(i, 0)):
                nums[index] = i
                index += 1

# revise - 02/19/2025